Library/Homebrew/shell_command.rb

```diff
@@ -0,0 +1,18 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Homebrew
+  module ShellCommand
+    extend T::Helpers
+
+    requires_ancestor { T.class_of(AbstractCommand) }
```

Comment:
```suggestion
    requires_ancestor { AbstractCommand }
```

Reply:
Sadly now it fails with another error:

```
shell_command.rb:12: Method `dev_cmd?` does not exist on `T::Class[Homebrew::ShellCommand]` https://srb.help/7003
    12 |      sh_cmd_path = "#{self.class.dev_cmd? ? "dev-cmd" : "cmd"}/#{self.class.command_name}.sh"
                                          ^^^^^^^^
  Got `T::Class[Homebrew::ShellCommand]` originating from:
    shell_command.rb:12:
    12 |      sh_cmd_path = "#{self.class.dev_cmd? ? "dev-cmd" : "cmd"}/#{self.class.command_name}.sh"
                               ^^^^^^^^^^
shell_command.rb:12: Method `command_name` does not exist on `T::Class[Homebrew::ShellCommand]` https://srb.help/7003
    12 |      sh_cmd_path = "#{self.class.dev_cmd? ? "dev-cmd" : "cmd"}/#{self.class.command_name}.sh"
                                                                                     ^^^^^^^^^^^^
  Got `T::Class[Homebrew::ShellCommand]` originating from:
    shell_command.rb:12:
    12 |      sh_cmd_path = "#{self.class.dev_cmd? ? "dev-cmd" : "cmd"}/#{self.class.command_name}.sh"
                                                                          ^^^^^^^^^^
  There is a singleton class method with the same name:
    test/abstract_command_spec.rb:50: Defined here
    50 |            def self.command_name = "t-a-c"
                    ^^^^^^^^^^^^^^^^^^^^^
    Either:
    - use `.class` to call it, or
    - remove `self.` from its definition to make it an instance method
  Autocorrect: Use `-a` to autocorrect
    shell_command.rb:12: Insert `.class`
    12 |      sh_cmd_path = "#{self.class.dev_cmd? ? "dev-cmd" : "cmd"}/#{self.class.command_name}.sh"
                                                                                    ^
```

Any ideas?

Reply:
Ah I think that covers the case where you can do `ShellCommand.new.run` which obviously we won't be but there isn't a way to state that with `requires_ancestor`.

Try add `T.bind(self, AbstractCommand)` to the top of `def run`.

Reply:
That worked. Thanks!